torch/optim/adam.py

```diff
@@ -671,7 +671,14 @@ def _fused_adam(
         if lr_dict is not None and device not in lr_dict:
             lr_dict[device] = lr.to(device=device, non_blocking=True)  # type: ignore[union-attr]
             lr = lr_dict[device]
-        torch._foreach_add_(device_state_steps, 1)
+        if device.type == "mps":  # type: ignore[union-attr]
```

Comment:
I have a PR that will allow this to work for MPS! :D  https://github.com/pytorch/pytorch/pull/127412

So let's land this PR after and we should not need this change.